packages/core/contracts/financial-templates/common/FundingRatePayer.sol

```diff
@@ -72,20 +72,20 @@ abstract contract FundingRatePayer is FeePayer, PerpetualInterface {
         );
         FixedPoint.Unsigned memory collateralPool = _pfc();
 
-        // If contract has no PfC, then cannot pay any fees.
-        require(!collateralPool.isEqual(0), "PfC is 0");
-        // The fee must be non-zero.
-        require(!amount.isEqual(0), "Funding fee is 0");
-        // The fee must be < available collateral.
-        require(collateralPool.isGreaterThan(amount), "Funding fee is more than PfC");
+        // If amount to pay or collateral pool is 0, then return early.
```

Comment:
One minor note I just realized... we don't pay regular fees before paying out funding rate fees. Does this have any weird implications? Previously, the assumption was that you pay regular fees before doing anything to change the pfc, but if we were to do that in this case, the calculation in the FRS would be slightly inaccurate since the regular fees would change the pfc before calling this method.

Reply:
Great point. What if withdrawFundingRateFees(amount) only withdrew the regular-fee-adjusted amount of funding rate fees to the store, and returned the adjusted amount to the caller. And then the Store used this adjusted amount to pay out rewards.

Reply:
This would still allow the proposers reward to be a ”% of the PfC” but comments and perhaps variable names should be changed to reflect that this is the "fee adjusted Pfc"

Reply:
Thinking more about this, I think both ways of doing this (the way you have it now vs changing it) are equally "correct."

The way you have it now, the regular fees and the proposal fees are both assessed on the "pre-transaction" collateral. That makes the percentages effectively additive. Although it may seem like the pfc we're using is "stale," I think there's a way of looking at it where we're just choosing not to "compound" these deductions. This implies that the calculations are done simultaneously without being ordered. I think that might be the more sensible choice here.

To pay the fees first and then apply the proposal reward percentage would be effectively making the reward a percentage of the post-fee pfc, which means that we've effectively "compounding" the deductions. This would result in a slightly smaller payment. This implies an ordering on the calculations whereby the regular fees are paid first and then the proposal deduction is applied in a later step. This also seems fairly sensible.

I think the only thing we _must_ do is ensure that regular fees are paid during `withdrawFundingRateFees`.

Reply:
I agree with your analysis, for implementation simplcity let's add the `fees()` modifier to `withdrawFundingRateFees`